pilot/pkg/model/push_context.go

```diff
@@ -750,7 +750,8 @@ func (ps *PushContext) UpdateMetrics() {
 }
 
 // It is called after virtual service short host name is resolved to FQDN
-func virtualServiceDestinations(v *networking.VirtualService) map[string]sets.Set[int] {
+// It filters virtualDestinations by using configNamespace, when the value is empty string, then filtering is disabled
+func virtualServiceDestinations(v *networking.VirtualService, configNamespace string) map[string]sets.Set[int] {
```

Comment:
I am always against using "" and other nonempty string to distinguish a semantic diff.

Prefer add a wrapper function,, should share most of the code